api/src/services/__tests__/competenciesService.ts

```diff
@@ -62,4 +64,31 @@ describe('competenciesService', () => {
       });
     });
   });
+
+  describe('updateCompetency', () => {
+    it('should update the specified/selected competency', async () => {
+      const id = 2;
+      const label = 'label';
+      const description = 'description';
+      mockQuery(
+        'update `competencies` set `label` = ?, `description` = ? where `id` = ?',
+        [label, description, id],
+        1
+      );
+      await updateCompetency(id, label, description);
+    });
+  });
+
+  describe('authorizeCompetencyUpdate', () => {
+    it('should authorize the specific competency update ', async () => {
+      const principalId = 2;
+      const competencyId = 3;
+      mockQuery(
+        'select `id` from `competencies` where `id` = ? and `principal_id` = ?',
+        [competencyId, principalId],
+        [competencyId]
```

Comment:
I just took a look at the knex docs for `update` and it looks like it could return a number of different things, but just an array with the `id` of the updated record isn't one of the options.

https://knexjs.org/#Builder-update

Can you confirm that this matches with the actual behaviour when the app is running? If it's different, this test could be confusing or may permit regressions.

Reply:
The mock query output now matches the authorized competency update function output.